### PR TITLE
Optimize filesystem bootstrapper performance

### PR DESCRIPTION
**Pull request by xichen2020**
cc @robskillington @martin-mao 

This reduces the time to bootstrap 48 hours of data from disk on a single node from 30 mins to 3-4 mins.

### CODE REVIEW

File: storage/bootstrap/bootstrapper/fs/source.go [hunk 1]
```diff
@@ -95,76 +107,196 @@ func (s *fileSystemSource) shardAvailability(
...
+	go s.enqueueReaders(namespace, shardsTimeRanges, readersCh)
+	result := s.bootstrapFromReaders(readersCh)
+
+	end := time.Now()
+
+	s.log.WithFields(
+		xlog.NewLogField("namespace", namespace),
+		xlog.NewLogField("numShards", len(shardsTimeRanges)),
+		xlog.NewLogField("duration", end.Sub(start).String()),
+	).Info("finished reading data from filesystem for bootstrapping")
```
Comment (robskillington) [inline storage/bootstrap/bootstrapper/fs/source.go]: Should we keep these more detailed logs?  If so perhaps we should make them more generic and do it one layer high in the bootstrap process?

File: storage/bootstrap/process.go [hunk 2]
```diff
@@ -67,18 +67,22 @@ func (b *bootstrapProcess) Run(
 			shardsTimeRanges[s] = r
 		}
 
-		b.log.WithFields(
+		logFields := []xlog.LogField{
 			xlog.NewLogField("namespace", namespace),
 			xlog.NewLogField("numShards", len(shards)),
 			xlog.NewLogField("from", window.Start.String()),
 			xlog.NewLogField("to", window.End.String()),
 			xlog.NewLogField("length", window.End.Sub(window.Start).String()),
```
Comment (robskillington) [inline storage/bootstrap/process.go]: Should we add the name of the bootstrapper we're calling somehow?

File: storage/namespace.go [hunk 3]
```diff
@@ -225,6 +229,8 @@ func (n *dbNamespace) Bootstrap(
 	}
 	n.metrics.bootstrap.Success.Inc(1)
 
+	n.log.WithFields(namespaceLogField).Info("finished reading external data")
```
Comment (robskillington) [inline storage/namespace.go]: Probably don't need the log statements in this file yeah?

File: storage/bootstrap/bootstrapper/fs/source.go (hunk 1 above)
Comment (xichen2020) [inline storage/bootstrap/bootstrapper/fs/source.go]: meh, they are somewhat useful but for now i'll move logging into the bootstrap manager

File: storage/namespace.go (hunk 3 above)
Comment (xichen2020) [inline storage/namespace.go]: It does provide timing breakdown of the different stages during bootstrap, will move it to bootstrap manager for now

File: storage/bootstrap/process.go (hunk 2 above)
Comment (xichen2020) [inline storage/bootstrap/process.go]: Sure thing

**Pull request merged**